### PR TITLE
initial cross-build to scala.js

### LINKED ISSUE #57: Support for Scala.js
I think this library can support Scaka.js because Shapeless supports it.

I am not sure if thats possible but it would be very nice

### PR DESCRIPTION
**Pull request by andreaTP**
Fix #57 
This is an initial port to Scala.Js.

To note:
the env variable of the seed is computed at compile time (to be tested).

### CODE REVIEW

File: js/src/main/scala/com/danielasfregola/randomdatagenerator/RandomDataGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package com.danielasfregola.randomdatagenerator
+
+import com.danielasfregola.randomdatagenerator.utils.{SeedDetector, ShapelessLike}
+import org.scalacheck._
+
+// import scala.reflect.runtime.universe._
```
Comment (DanielaSfregola) [inline js/src/main/scala/com/danielasfregola/randomdatagenerator/RandomDataGenerator.scala]: remove

File: varenv/shared/src/main/scala/com/danielasfregola/randomdatagenerator/macro/SeedVariable.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox.Context
+
+import scala.util.Properties
+
+object SeedVariable {
+
+  val SeedVariableName = "RANDOM_DATA_GENERATOR_SEED"
+
+  def nameImpl(c: Context)(): c.Expr[Option[String]] = {
```
Comment (DanielaSfregola) [inline varenv/shared/src/main/scala/com/danielasfregola/randomdatagenerator/macro/SeedVariable.scala]: remove as it doesn't work

**Review by DanielaSfregola — CHANGES_REQUESTED** [review]
Changes coming soon!

File: varenv/shared/src/main/scala/com/danielasfregola/randomdatagenerator/macro/SeedVariable.scala (hunk 2 above)
Comment (andreaTP) [reply varenv/shared/src/main/scala/com/danielasfregola/randomdatagenerator/macro/SeedVariable.scala]: removed

**Comment by andreaTP** [comment]
@DanielaSfregola 1 hour plane delay made it.
I'm happy with it now.
:-)

**Comment by andreaTP** [comment]
the approach on the env variable is now as follows on Scala.Js:
try to access the env assuming you are using Node (that is the most common case), fallback to None in all other cases.

**Comment by DanielaSfregola** [comment]
> @DanielaSfregola 1 hour plane delay made it.
> I'm happy with it now.
> :-)

I have never been happier of a plane delay! 😄

File: js/src/main/scala/com/danielasfregola/randomdatagenerator/utils/SeedVariable.scala [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.danielasfregola.randomdatagenerator.utils
+
+import scala.util.Try
+import scala.scalajs.js
+
+object SeedVariable {
+
+  protected[utils] lazy val name = Try {
+    val fromEnv = js.Dynamic.global.process.env.selectDynamic(SeedDetector.SeedVariableName)
```
Comment (DanielaSfregola) [inline js/src/main/scala/com/danielasfregola/randomdatagenerator/utils/SeedVariable.scala:9]: NICE!!!

File: jvm/src/main/scala/com/danielasfregola/randomdatagenerator/utils/SeedVariable.scala [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.danielasfregola.randomdatagenerator.utils
+
+import scala.util.Properties
+
+object SeedVariable {
+
+  protected[utils] lazy val name = Properties.envOrNone(SeedDetector.SeedVariableName)
```
Comment (DanielaSfregola) [inline jvm/src/main/scala/com/danielasfregola/randomdatagenerator/utils/SeedVariable.scala:7]: `name` is confusing, I'll change it in another PR

**Review by DanielaSfregola — APPROVED** [review]
LGTM!!!!

**Pull request merged**